### PR TITLE
Android fixes

### PR DESCRIPTION
**Pull request by mbender74**
Fixes for FragmentManager: No view found for id 0x2 (unknown) for fragment SupportMapFragment #282
and corrections now the mapView works as expected
Feat: added code for offlineOverlay image tiles like my PR soon to be done for iOS for offlineOverlay map features....

### CODE REVIEW

File: android/src/ti/map/ViewProxy.java [hunk 1]
```diff
@@ -95,6 +97,7 @@ public ViewProxy()
 	{
 		super();
 		preloadRoutes = new ArrayList<RouteProxy>();
+		// defaultValues.put(TiC.PROPERTY_FRAGMENT_ONLY, true);
```
Comment (m1ga) [inline android/src/ti/map/ViewProxy.java]: remove the comment if not needed

File: android/src/ti/map/ViewProxy.java [hunk 2]
```diff
@@ -110,7 +113,14 @@ public ViewProxy()
 	@Override
 	public TiUIView createView(Activity activity)
 	{
-		return new TiUIMapView(this, activity);
+
+		// setProperty(TiC.PROPERTY_FRAGMENT_ONLY, true);
```
Comment (m1ga) [inline android/src/ti/map/ViewProxy.java]: remove the comment if not needed

File: android/src/ti/map/ViewProxy.java [hunk 3]
```diff
@@ -110,7 +113,14 @@ public ViewProxy()
...
+
+		TiUIView mapview = new TiUIMapView(this, activity);
+
+		// ((TiUIMapView) view).createMapFragment();
```
Comment (m1ga) [inline android/src/ti/map/ViewProxy.java]: remove the comment if not needed

File: android/src/ti/map/ViewProxy.java [hunk 4]
```diff
@@ -1226,7 +1238,8 @@ private float handleGetZoomLevel()
 	public void setLocation(Object location)
 	{
 		if (location instanceof HashMap) {
-			HashMap dict = (HashMap) location;
+			HashMap<String, Object> dict = new HashMap<>();
+			dict.put("", location);
```
Comment (m1ga) [inline android/src/ti/map/ViewProxy.java]: empty key

**Comment by m1ga** [comment]
why did you do the PR on my fork :smile: ?

**Comment by mbender74** [comment]
```
var mapView = mapModule.createView({
			id:'mapview',
			userLocation: true,
			minZoom:14,
			maxZoom:18,
			mapType: Map.SATELLITE_TYPE,
			animate: false,
			region: {
				latitude: 49.54119,
				longitude: 7.33233,
				latitudeDelta: parseFloat((parseFloat(49.5487)-parseFloat(49.5300))/2),
				longitudeDelta: parseFloat((parseFloat(7.3453)-parseFloat(7.3170))/2),
			},
			offlineOverlay: {
				tilesPath:'/maps/tiles',
				tileExtension:'jpg',
				tileSize:256,
				maxRegion: {
					minLat:49.5300,
					maxLat:49.5487,
					minLong:7.3170,
					maxLong:7.3453,
					maxZoom:18,
					minZoom:14
				}
			},
			pitchEnabled: true,
			rotateEnabled: true,
			compassEnabled: true,
			showsBuildings: true,
			showsPointsOfInterest: true,
			top:0,
			bottom:0,
			left:0,
			right:0,
			width: Ti.UI.FILL,
			height: Ti.UI.FILL,
		});
```
[maps.zip](https://github.com/m1ga/ti.map/files/8927358/maps.zip)

**Comment by mbender74** [comment]
because you need to test first

**Comment by mbender74** [comment]
will do on main repo if found for good :-)

**Comment by mbender74** [comment]
this fragment bug, also with your further improvements, which did not take any properties, just rendering the world.... take me hours over hours... this morning, I finally get it to work :-)

File: android/src/ti/map/ViewProxy.java (hunk 1 above)
Comment (mbender74) [reply android/src/ti/map/ViewProxy.java]: ah, right

File: android/src/ti/map/ViewProxy.java (hunk 4 above)
Comment (mbender74) [reply android/src/ti/map/ViewProxy.java]: well, that way no warnings from compiler because of wrong hashmap type

Comment (mbender74) [reply android/src/ti/map/ViewProxy.java]: if that has to be diffrent, I don´t know :-D

**Review by mbender74 — COMMENTED** [review]
comments removed

**Comment by mbender74** [comment]
in the maps.zip are demo map tiles, that will show the offlineOverlay feature

**Comment by m1ga** [comment]
offline tiles work fine. Make sure to add the properties to the docs (https://github.com/m1ga/ti.map/blob/master/apidoc/View.yml).
Can't test the other issue since it wasn't my bug.
I've created this issue: https://github.com/tidev/ti.map/issues/480 which still won't work and it will only show one map.

> because you need to test first

but my fork is not up to date (only one PR with iOS stuff) and I can test it from your fork or the official repo too :smile: This only makes me double the work since I have to test it here and in the tidev repo. I rather prefer doing the PR in the main repo right way or no PR and just use your fork until its ready :wink:

File: android/src/ti/map/ViewProxy.java (hunk 4 above)
Comment (m1ga) [reply android/src/ti/map/ViewProxy.java]: the `warning: [rawtypes] found raw type: HashMap` warnings have been there the whole time. If you make a `ti clean` and build again you'll see many other warnings like
```
HashMap centerOffsetProperty = (HashMap) anno.getProperty(MapModule.PROPERTY_CENTER_OFFSET);
[ERROR] [GRADLE]                                ^
[ERROR] [GRADLE]   missing type arguments for generic class HashMap<K,V>
```
but it still builds fine. I would rather keep the part from before and we fix those warnings at another point. I've not tested setLocation but `dict.put("", location);` doesn't look right either :rofl:

**Comment by mbender74** [comment]
ok, I will also check if I can fix https://github.com/tidev/ti.map/issues/480
and then I will push to official....

File: android/src/ti/map/CustomMapFragment.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
+package ti.map;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
```
Comment (m1ga) [inline android/src/ti/map/CustomMapFragment.java]: unused imports

File: android/src/ti/map/CustomMapFragment.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+		return fragment;
+	}
+
+	// @Override
+	// public View onCreateView(LayoutInflater arg0, ViewGroup arg1, Bundle arg2) {
+	//       Log.e(TAG, " onCreateView");
+	//       View v = super.onCreateView(arg0, arg1, arg2);
+	//       return v;
+	// }
```
Comment (m1ga) [inline android/src/ti/map/CustomMapFragment.java]: remove if not needed

File: android/src/ti/map/TiUIMapView.java [hunk 7]
```diff
@@ -85,21 +104,93 @@ public class TiUIMapView extends TiUIFragment
...
+	private String tileExtension;
+	private Integer tileSize;
+	private boolean useOfflineOverlay = false;
+	private boolean maxBoundsSet = false;
+	private TiViewProxy myProxy;
+	private float minZoomPreference = 14;
+	private float maxZoomPreference = 20;
+	private LatLngBounds boundingRegion;
+	private Context mainContext;
+	private GroundOverlay overlayBackground;
```
Comment (m1ga) [inline android/src/ti/map/TiUIMapView.java]: never used

File: android/src/ti/map/TiUIMapView.java [hunk 8]
```diff
@@ -85,21 +104,93 @@ public class TiUIMapView extends TiUIFragment
...
+	private Integer backgroundColor = null;
+	private boolean transactionCommitted = false;
```
Comment (m1ga) [inline android/src/ti/map/TiUIMapView.java]: unused

File: android/src/ti/map/TiUIMapView.java [hunk 9]
```diff
@@ -85,21 +104,93 @@ public class TiUIMapView extends TiUIFragment
...
+			gOptions.liteMode(true);
+		}
+
+		if (this.backgroundColor != null) {
+			gOptions.backgroundColor(this.backgroundColor);
+		}
+		mapView = new MapView(proxy.getActivity(), gOptions);
+		container.addView(mapView);
+		mapView.onCreate(Bundle.EMPTY);
+		mapView.onResume();
```
Comment (m1ga) [inline android/src/ti/map/TiUIMapView.java:152]: is that `onResume` needed?

File: android/src/ti/map/TiUIMapView.java [hunk 10]
```diff
@@ -1418,7 +1532,25 @@ public void onMapLoaded()
...
+				Marker marker = timarker.getMarker();
+				if (map != null && marker != null && marker.isInfoWindowShown()) {
+					Point markerPoint = map.getProjection().toScreenLocation(marker.getPosition());
+					infoWindow.analyzeTouchEvent(ev, markerPoint, selectedAnnotation.getIconImageHeight());
+				}
+			}
+		}
+		return false;
+	}
+
```
Comment (m1ga) [inline android/src/ti/map/TiUIMapView.java]: why did you move that function? Unnecessary change :smile:

File: android/src/ti/map/TilesOfflineTileProvider.java [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+import org.appcelerator.kroll.common.TiMessenger;
+import org.appcelerator.titanium.TiApplication;
+import org.appcelerator.titanium.TiC;
+import org.appcelerator.titanium.io.TiBaseFile;
+import org.appcelerator.titanium.io.TiFileFactory;
+import org.appcelerator.titanium.io.TiFileProvider;
+import org.appcelerator.titanium.proxy.TiViewProxy;
+import org.appcelerator.titanium.util.TiFileHelper2;
+import org.appcelerator.titanium.util.TiRHelper;
+import org.appcelerator.titanium.view.TiDrawableReference;
```
Comment (m1ga) [inline android/src/ti/map/TilesOfflineTileProvider.java]: only 
```
import android.graphics.Bitmap;

import com.google.android.gms.maps.model.Tile;
import com.google.android.gms.maps.model.TileProvider;

import org.appcelerator.titanium.proxy.TiViewProxy;
import org.appcelerator.titanium.view.TiDrawableReference;

import java.io.ByteArrayOutputStream;
``` 
are used

File: android/src/ti/map/ViewProxy.java [hunk 12]
```diff
@@ -110,7 +112,8 @@ public ViewProxy()
 	@Override
 	public TiUIView createView(Activity activity)
 	{
-		return new TiUIMapView(this, activity);
+		TiUIView view = new TiUIMapView(this);
+		return view;
```
Comment (m1ga) [inline android/src/ti/map/ViewProxy.java]: remove the variable and just return `new...`

**Review by m1ga — COMMENTED** [review]
left some comments. In case you don't use Android Studio: use it :smile: makes it easier to spot unused variables and imports!
